jsign-crypto/src/test/java/net/jsign/jca/VenafiSigningServiceTest.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright 2025 Ivan Wallis
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.jsign.jca;
+
+import java.io.FileReader;
+import java.security.GeneralSecurityException;
+import java.security.KeyStoreException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Collections;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static net.jadler.Jadler.*;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+public class VenafiSigningServiceTest {
+
+    @Before
+    public void setUp() {
+        initJadler().withDefaultResponseStatus(404);
+    }
+
+    @After
+    public void tearDown() {
+        closeJadler();
+    }
+
+    @Test
+    public void testGetAliases() throws Exception {
+        SigningService service = new VenafiSigningService("http://localhost:" + port(), new VenafiCredentials("username", "password", null));
+        List<String> aliases = service.aliases();
+
+        assertEquals("aliases", Collections.emptyList(), aliases);
+    }
+
+    @Test
+    public void testGetCertificateChain() throws Exception {
+        onRequest()
```

Comment:
The test server should also mock the requests to /vedauth/authorize/oauth, otherwise the authentication fails due to a 404 error.

Reply:
Can you provide more guidance?

Reply:
For each test it's necessary to mock the authentication request by adding something like this:

```
onRequest("/vedauth/authorize/oauth")
    .havingMethodEqualTo("POST")
    .respond()
    .withStatus(200)
    .withBody(new FileReader("target/test-classes/services/venafi-authorize.json"));
```

This will also require a new JSON file in `jsign-crypto/src/test/resources/services/` containing the server response to this API call.